src/worker/util/advStats.basketball.ts

```diff
@@ -154,6 +157,323 @@ const calculatePER = (players: any[], teamsInput: Team[], league: any) => {
+		0.422,
+		0.103,
+		0.0,
+		0.294,
+		0.097,
+		-0.439,
+	];
+
+	let BPM: number[] = [];
+	let OBPM: number[] = [];
+
+	for (let i = 0; i < players.length; i++) {
+		const p = players[i].stats;
+		const role = playerRole[i];
+		const pos = playerPos[i];
+
+		const poss = playerPoss[i];
+		const pts100 = (adjPts[i] / poss) * 100;
+		const fga100 = (p.fga / poss) * 100;
+		const fta100 = (p.fta / poss) * 100;
+		const tp100 = (p.tp / poss) * 100;
+		const ast100 = (p.ast / poss) * 100;
+		const stl100 = (p.stl / poss) * 100;
+		const blk100 = (p.blk / poss) * 100;
+		const pf100 = (p.pf / poss) * 100;
+
+		const to100 = (p.tov / poss) * 100;
+		const orb100 = (p.orb / poss) * 100;
+		const drb100 = (p.drb / poss) * 100;
+		const trb100 = (p.trb / poss) * 100;
+		const minp = playerMin[i];
+
+		const coeffsBPM: number[] = [];
+		const coeffsORBPM: number[] = [];
+
+		const interBPM =
+			pos < 3
+				? ((3 - pos) / 2) * -0.818
+				: ((pos - 3) / 2) * 0 + ((5 - pos) / 2) * 0 + 1.387 * (role - 3);
+		const interORBPM =
+			pos < 3
+				? ((3 - pos) / 2) * -1.698
+				: ((pos - 3) / 2) * 0 + ((5 - pos) / 2) * 0 + 0.43 * (role - 3);
+
+		for (let j = 0; j < coeffsBPM1.length; j++) {
+			const posB = j === 1 || j === 2 ? role : pos;
+
+			coeffsBPM[j] =
+				((5 - posB) / 4) * coeffsBPM1[j] + ((posB - 1) / 4) * coeffsBPM5[j];
+			coeffsORBPM[j] =
+				((5 - posB) / 4) * coeffsORBPM1[j] + ((posB - 1) / 4) * coeffsORBPM5[j];
+		}
+
+		const scoring =
+			pts100 * coeffsBPM[0] +
+			fga100 * coeffsBPM[1] +
+			fta100 * coeffsBPM[2] +
+			tp100 * coeffsBPM[3];
+		const scoring2 =
+			pts100 * coeffsORBPM[0] +
+			fga100 * coeffsORBPM[1] +
+			fta100 * coeffsORBPM[2] +
+			tp100 * coeffsORBPM[3];
+
+		const ballhandle = ast100 * coeffsBPM[4] + to100 * coeffsBPM[5];
+		const ballhandle2 = ast100 * coeffsORBPM[4] + to100 * coeffsORBPM[5];
+
+		const defense =
+			orb100 * coeffsBPM[6] +
+			drb100 * coeffsBPM[7] +
+			trb100 * coeffsBPM[8] +
+			stl100 * coeffsBPM[9] +
+			blk100 * coeffsBPM[10] +
+			pf100 * coeffsBPM[11];
+		const defense2 =
+			orb100 * coeffsORBPM[6] +
+			drb100 * coeffsORBPM[7] +
+			trb100 * coeffsORBPM[8] +
+			stl100 * coeffsORBPM[9] +
+			blk100 * coeffsORBPM[10] +
+			pf100 * coeffsORBPM[11];
+		const rawBPM = scoring + ballhandle + defense + interBPM;
+		const rawOBPM = scoring2 + ballhandle2 + defense2 + interORBPM;
+
+		BPM[i] = rawBPM;
+		OBPM[i] = rawOBPM;
+
+		teamBPM[players[i].tid] += rawBPM * minp;
+		teamOBPM[players[i].tid] += rawOBPM * minp;
+	}
+	const teamAdjBPM: number[] = [];
+	const teamAdjOBPM: number[] = [];
+
+	for (let i = 0; i < teams.length; i++) {
+		teamAdjBPM[i] = (tmRate[i] - teamBPM[i]) / 5;
+		teamAdjOBPM[i] = (ofRate[i] - teamOBPM[i]) / 5;
+	}
+	const DBPM: number[] = [];
+	const VORP: number[] = [];
+	for (let i = 0; i < players.length; i++) {
+		BPM[i] += teamAdjBPM[players[i].tid];
+		OBPM[i] += teamAdjOBPM[players[i].tid];
+		DBPM[i] = BPM[i] - OBPM[i];
+		VORP[i] = playerMin[i] * BPM[i];
```

Comment:
https://www.basketball-reference.com/about/bpm2.html says VORP is `[BPM - (-2.0)] * (% of possessions played) * (team games/82)`, looks like this is missing the constant (just scales it up) and the team games part (makes it so that players don't immediately have huge VORPs after 1 game in the season).

I'll fix it, just wanted to make sure this wasn't intentional